### PR TITLE
Expose agency

### PR DESCRIPTION
**Pull request by lsetiawan**
This PR addresses #162. `agencyName` and `agencyCode` should show up when organizationtype matches either `Government agency` or `Research agency`.

### CODE REVIEW

File: wof/examples/flask/odm2/timeseries/odm2_timeseries_dao.py [hunk 1]
```diff
@@ -68,10 +68,19 @@ def get_site_by_code(self, site_code):
...
             s = self.db_session.query(odm2_models.Sites). \
                 filter(odm2_models.Sites.SamplingFeatureCode == site_code).one()
+
+            aff = self.db_session.query(odm2_models.Affiliations). \
+                join(odm2_models.ActionBy). \
+                join(odm2_models.Actions). \
+                join(odm2_models.FeatureActions). \
+                join(odm2_models.Sites). \
+                filter(odm2_models.Sites.SamplingFeatureCode == s.SamplingFeatureCode). \
+                filter(odm2_models.ActionBy.IsActionLead == True).first()
```
Comment (emiliom) [inline wof/examples/flask/odm2/timeseries/odm2_timeseries_dao.py]: The criteria `odm2_models.ActionBy.IsActionLead == True` may be too strict, in this context. Either remove it (that *could* lead to problems in some situations), or add a comment next to the line in the code, for future reference.

File: wof/examples/flask/odm2/timeseries/sqlalch_odm2_models.py [hunk 2]
```diff
@@ -204,7 +207,7 @@ def __init__(self, aff_obj):
 
         if aff_obj.OrganizationObj is not None:
             self.Organization = aff_obj.OrganizationObj.OrganizationName
-            self.OrganizationCode = aff_obj.OrganizationObj.OrganizationCode
+            self.SourceCode = aff_obj.OrganizationObj.OrganizationCode
             self.SourceDescription = aff_obj.OrganizationObj.OrganizationDescription
```
Comment (emiliom) [inline wof/examples/flask/odm2/timeseries/sqlalch_odm2_models.py]: > `self.SourceCode = aff_obj.OrganizationObj.OrganizationCode`

This is wrong, at least according to the interpretation I laid out in https://github.com/ODM2/WOFpy/issues/162#issue-248940883
There, I suggested it should be:
`self.SourceCode = aff_obj.AffiliationID`
or even more explicitly:
`self.SourceCode = self.SourceID`

Thoughts?

Comment (lsetiawan) [inline wof/examples/flask/odm2/timeseries/sqlalch_odm2_models.py]: > self.SourceCode = self.SourceID

Should be fine.

**Comment by emiliom** [comment]
Thanks!! Awesome. See my comments and let me know what you think, before we merge this PR.

Did you test this along the lines of what we discussed today?

Also, for clarity, this PR is not only enhancing the core/common wofpy code to fully support agency attributes; it's also **specifically implementing it in the ODM2 timeseries DAO only**. Nothing wrong with that. Just making this scope explicit here.

**Comment by emiliom** [comment]
Also, I just noticed that Travis-CI failed but AppVeyor succeeded?! Do you know why?

**Comment by lsetiawan** [comment]
> Did you test this along the lines of what we discussed today?

Yepp all is tested.

> Also, I just noticed that Travis-CI failed but AppVeyor succeeded?! Do you know why?

Weird.. Travis says this is because there's no Affiliation:

``` AttributeError: 'NoneType' object has no attribute 'OrganizationObj'``` I'll get error catching soon. 

This PR is not ready to merge until further notice and fixes.

**Comment by emiliom** [comment]
> Yepp all is tested.

Great.

> This PR is not ready to merge until further notice and fixes.

Ok

**Comment by lsetiawan** [comment]
@emiliom Please review further my latest changes and merge when you think they're good. Thanks.

**Comment by emiliom** [comment]
Looks good. Merging now.

**Pull request merged**